include/EASTL/heap.h

```diff
@@ -87,7 +87,7 @@ namespace eastl
 	/// This function requires that the value argument refer to a value
 	/// that is currently not within the heap.
 	///
-	template <typename RandomAccessIterator, typename Distance, typename T, typename Compare>
+	template <typename RandomAccessIterator, typename Distance, typename T, typename Compare, typename = typename eastl::enable_if<eastl::is_copy_constructible<typename eastl::iterator_traits<RandomAccessIterator>::value_type>::value>::type>
```

Comment:
similar issue here... this is function body using copy-assignment.

also, please add unit tests to validate this behaviour.